routes/_pages/settings/instances/[instanceName].html

```diff
@@ -159,6 +191,15 @@ <h2>Theme:</h2>
         selectedTheme: instanceThemes[instanceName] || 'default'
       })
       await updateVerifyCredentialsForInstance(instanceName)
+      await updatePushSubscriptionForInstance(instanceName)
+
+      const form = this.refs.pushNotificationsForm
+      const { pushSubscription } = this.store.get()
+
+      form.elements.follow.checked = pushSubscription && pushSubscription.alerts && pushSubscription.alerts.follow
+      form.elements.favourite.checked = pushSubscription && pushSubscription.alerts && pushSubscription.alerts.favourite
+      form.elements.reblog.checked = pushSubscription && pushSubscription.alerts && pushSubscription.alerts.reblog
+      form.elements.mention.checked = pushSubscription && pushSubscription.alerts && pushSubscription.alerts.mention
```

Comment:
There should be a way to bind this to the store data without needing to use `refs`. The checkboxes for themes use a similar system.

Reply:
I tried to avoid using `bind:checked` since the changes are not applied instantly (like the themes), since we need to perform a network request that may fail (handling that would mean having to revert the state of the store, which implies keeping track of the previous state of `store.pushSubscription`).

Reply:
Hmm, in that case I would still prefer to handle it with state than with direct DOM manipulation. In the past for these kinds of "optimistic" UI updates, I have used state overrides. For instance, you can set a boolean value on the local component state that overrides the store and just updates the checked state. I'll make the modification so you can see what I mean :)

Reply:
Hmm, I am playing around with this, and the code is pretty gnarly. :( Maybe it will be best to just merge your PR as-is, since everything else looks fantastic to me. (Frankly I am amazed you managed to figure out my idiomatic code as much as you did! I didn't document everything nearly as well as I could have.)

Reply:
Apart from my unfamiliarity with Svelte I didn't have major issues navigating the code, looks very well structured to me :)